From 2d71f6ea0843af1e218113ac8309b3966fcffc27 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 24 Apr 2018 14:28:33 +0100
Subject: [PATCH 309/321] BCM270X_DT: Add sdio_overclock parameter

The Arasan MMC interface is used on some RPis to drive the SDIO
link to the WiFi controller. The downstream bcm2835-mmc driver,
like the bcm2835-sdhost driver, can be over- (or under-) clocked.
Add a common parameter - sdio_overclock - to all DTBs to control it.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/bcm2708-rpi.dtsi | 1 +
 arch/arm/boot/dts/overlays/README  | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/bcm2708-rpi.dtsi b/arch/arm/boot/dts/bcm2708-rpi.dtsi
index d105d4e208f2..fe2694a1cbdc 100644
--- a/arch/arm/boot/dts/bcm2708-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2708-rpi.dtsi
@@ -115,6 +115,7 @@
 		sd_force_pio = <&sdhost>,"brcm,force-pio?";
 		sd_pio_limit = <&sdhost>,"brcm,pio-limit:0";
 		sd_debug     = <&sdhost>,"brcm,debug";
+		sdio_overclock = <&mmc>,"brcm,overclock-50:0";
 		axiperf      = <&axiperf>,"status";
 	};
 };
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index ca6995142322..81e61c0d6bc2 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -139,6 +139,9 @@ Params:
 
         sd_debug                Enable debug output from SD driver (default off)
 
+        sdio_overclock          Clock (in MHz) to use when the MMC framework
+                                requests 50MHz for the SDIO/WiFi interface.
+
         tx_lpi_timer            Set the delay in microseconds between going idle
                                 and entering the low power state (default 600).
                                 Requires EEE to be enabled - see "eee".
-- 
2.16.1

